MyList/src/main/java/com/example/mylist/MyList.java

```diff
@@ -0,0 +1,451 @@
+package com.example.mylist;
+
+import java.util.*;
+
+public class MyList<E> implements List<E>{
+
+    private static Object[] elementData;
+
+    private final int DEFAULT_CAPACITY = 5;
+
+    private int size;
+
+
+    //생성자 시작========================================================================
+
+    //1. 기본생성자
+    public MyList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    //2. 사이즈 지정생성자
+    public MyList(int n) {
+        if(n > 0) {
+            this.elementData = new Object[n];
+        }
+        if(n == 0) {
+            this.elementData = new Object[DEFAULT_CAPACITY];
+        }
+        if(n < 0) {
+            throw new IllegalArgumentException("Illegal Capacity: " + n);
+        }
+    }
+
+
+    //3. 컬렉션주입 생성자
+    public MyList(Collection<? extends E> c) {
+        elementData = c.toArray();
+        if(elementData.length != 0) {
+            size = elementData.length;
+        }
+        else {
+            this.elementData = new Object[DEFAULT_CAPACITY];
+        }
+    }
+
+    //테스트용 생성자
+    //아 테스트용 생성자 만들고 실제 생성자들은 private해도 될거같다
+    public MyList(String number, String number1, String number2) {
+        this.elementData = new Object[3];
+        elementData[0] = number;
+        elementData[1] = number1;
+        elementData[2] = number2;
+        size = 3;
+    }
+
+
+    //생성자 끝========================================================================
+
+
+    //add시작=========================================================================
+    @Override
+    public boolean add(E e) {
+        //꽉찬경우
+        if(size == elementData.length) {
+            grow();
+        }
+        elementData[size] = e;
+        size++;
+        return true;
+
+    }
+
+
+    // 배열의 크기를 두 배로 확장하는 메서드
+    private void grow() {
+        int oldCapacity = elementData.length;
+        int newCapacity = oldCapacity + (oldCapacity >> 1);
+        Object[] tobe = new Object[newCapacity];
+        System.arraycopy(elementData, 0, tobe, 0, size);
+        elementData = tobe;
+
+    }
+
+
+
+    @Override
+    public void add(int index, E element) {
+        if(index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if(size == elementData.length) {
+            grow();
+        }
+        //index부터 뒤로 한칸씩 미루기
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = element;
+        size++;
+    }
+
+
+    @Override
+    public boolean addAll(Collection<? extends E> c) {
+        //true/false반환하기위한 플래그 필요
+        boolean retrunValue = false;
+        //for문 돌때마다 플래그변수 대입되는게 불편...
+        // 근데 true에서 또 true대입하게되면 그냥 자바는 무시할까?
+        for(E o :c) {
+            if(add(o)) {
+                retrunValue = true;
+            }
+        }
+
+        return retrunValue;
+    }
```

Comment:
아예 변수를 안쓰고 하는 방법도 있습니다.

```suggestion
    @Override
    public boolean addAll(Collection<? extends E> c) {
        //true/false반환하기위한 플래그 필요
        boolean retrunValue = false;
        //for문 돌때마다 플래그변수 대입되는게 불편...
        // 근데 true에서 또 true대입하게되면 그냥 자바는 무시할까?
        for(E o :c) {
            if(add(o)) {
                return true;
            }
        }

        return false;
    }
```

Reply:
컬렉션 내부 요소들을 모두 추가해야되기때문에
변수를 선언하고 진행하였습니다
제가 잘못 생각하고있는걸수도있지만
처음에 저도 저렇게 시도하였다가
저렇게 시도하면 컬렉션요소 한번만 체크하고 바로 true가 리턴되니깐 컬렉션 다음 요소를 추가할수없어서
변수를 추가한거였는데...
제가 한번 다시 찾아보겠습니다!

Reply:
아하 그렇네요. 모든 요소를 추가하는것이니 저렇게 리턴하면 문제가 생기겠네요. 잘 캐치해주셨습니다 👍